src/venue-service/models/venue-details.ts

```diff
@@ -1,25 +1,37 @@
 import { checkRequiredProperty } from '../../utils';
-import { VenueAddress } from '../../content-service/models';
+import { AddressModel } from './address';
 import { SeatSettings } from './seat-settings';
 import { VenueDetailsApi } from '../typings';
+import { VenueTerminal } from './terminal';
+import { Facility } from './facility';
 
 export class VenueDetails {
   private readonly seatSettings: SeatSettings;
-  private readonly address: VenueAddress;
+  private readonly address: AddressModel;
   private readonly title: string;
   private readonly internalId: string;
   private readonly description: string;
   private readonly createdAt: string;
+  private readonly venueTerminals: VenueTerminal[];
+  private readonly facilities: Facility[];
+  private readonly transportAttributes: string[];
 
   constructor (venueDetails: VenueDetailsApi) {
     checkRequiredProperty(venueDetails, 'Venue details: api settings');
 
+    const { venueTerminals = [], facilities = [], transportAttributes = [] } = venueDetails;
+
     this.seatSettings = new SeatSettings(venueDetails.seatSettings);
```

Comment:
Add checks for seatSettings and address. I've tried to call the service and got the exception, while I expect to see the model with null in seatSettings

Reply:
i'm not sure i unserstand
_` I've tried to call the service and got the exception`_  - is that not right behaviour?

Reply:
No, it's not expected because a manual call to the service should throw an error only if the original service respond with error. In this case you got the response, but part of the data is not here, so the model mapper is wrong.

Reply:
done

Reply:
const methods = venueService.create(Environment.Qa);
  const chartDetails = await methods.getChartDetails('1587');
  const venueDetails = await methods.getDetails('138');
  const seatAttribute = await methods.getSeatAttributesBySeatIds('138', ['Stalls-B10']);
  const seatAttributes = await methods.getSeatAttributes('138', '2020-10-10', '19:30')
  console.log('Chart details ', chartDetails);
  console.log('Venue details', venueDetails);
  console.log('Seat Attribute', seatAttribute);
  console.log('Seat Attributes', seatAttributes);

fixed seatsettings and all requests works without exceptions
![image](https://user-images.githubusercontent.com/66787159/95961110-4c283800-0e0d-11eb-9c2d-49549cae1df4.png)